src/SonarScanner.MSBuild.PreProcessor/AnalysisConfigProcessing/AnalysisConfigGenerator.cs

```diff
@@ -59,8 +59,11 @@ public static AnalysisConfig GenerateFile(ProcessedArgs localSettings,
             SonarBinDir = buildSettings.SonarBinDirectory,
             SonarScannerWorkingDirectory = buildSettings.SonarScannerWorkingDirectory,
             SourcesDirectory = buildSettings.SourcesDirectory,
-            JavaExePath = string.IsNullOrWhiteSpace(localSettings.JavaExePath) ? resolvedJavaExePath : localSettings.JavaExePath, // the user-specified JRE overrides the resolved value
-            EngineJarPath = string.IsNullOrWhiteSpace(localSettings.EngineJarPath) ? resolvedEngineJarPath : localSettings.EngineJarPath, // the user-specified engine.jar overrides the resolved value
+            // When passing paths between begin and end step, they must be absolute paths
```

Comment:
This comment is missleading. `they must be absolute paths` is done later and not here.